android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestImpression.kt

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.fxsuggest
+
+import mozilla.appservices.suggest.Suggestion
+import mozilla.components.concept.awesomebar.AwesomeBar
+
+sealed interface FxSuggestImpression {
+    data class Amp(
+        val blockId: Long,
+        val reportingUrl: String,
+        val advertiser: String,
+        val iabCategory: String,
+    ) : FxSuggestImpression
+}
+
+internal fun Suggestion.toImpression(): FxSuggestImpression? =
+    when (this) {
+        is Suggestion.Amp -> FxSuggestImpression.Amp(
+            blockId = blockId,
+            reportingUrl = impressionUrl,
+            advertiser = advertiser,
+            iabCategory = iabCategory,
+        )
+        else -> null
+    }
+
+interface FxSuggestImpressionRecorder {
+    fun recordPendingImpression(impression: FxSuggestImpression, position: AwesomeBar.Suggestion.Position)
+    fun forgetPendingImpression(impression: FxSuggestImpression)
+}
```

Comment:
I think we can ditch this interface in favor of the callback strategy I discussed above to hopefully remove a layer of abstraction here.